src/Controller/Component/CrudComponent.php

```diff
@@ -649,7 +649,13 @@ protected function _loadListener($name)
                 throw new ListenerNotConfiguredException(sprintf('Listener "%s" is not configured', $name));
             }
 
+            // @todo: temporary hack to prevent `undefined index: className` error because `api` is in listeners list even though not enabled.
```

Comment:
This happens as soon as I add the JsonApiListener, regardless by itself or along with ApiListener, PaginationListener or any other combination. Somehow `api` always ends up in the list of listeners even though I did not enable it.

```
Could not find listener class:
Crud\Error\Exception\MissingListenerException

Notice (8): Undefined index: className [ROOT/vendor/friendsofcake/crud/src/Controller/Component/CrudComponent.php, line 657]
Code Context
//            }

$className = App::classname($config['className'], 'Listener', 'Listener');
Crud\Controller\Component\CrudComponent::_loadListener() - ROOT/vendor/friendsofcake/crud/src/Controller/Component/CrudComponent.php, line 657
Crud\Controller\Component\CrudComponent::_loadListeners() - ROOT/vendor/friendsofcake/crud/src/Controller/Component/CrudComponent.php, line 631
Crud\Controller\Component\CrudComponent::beforeFilter() - ROOT/vendor/friendsofcake/crud/src/Controller/Component/CrudComponent.php, line 190
Cake\Event\EventManager::_callListener() - CORE/src/Event/EventManager.php, line 422
Cake\Event\EventManager::dispatch() - CORE/src/Event/EventManager.php, line 391
Cake\Controller\Controller::dispatchEvent() - CORE/src/Event/EventDispatcherTrait.php, line 78
Cake\Controller\Controller::startupProcess() - CORE/src/Controller/Controller.php, line 497
Cake\Http\ActionDispatcher::_invoke() - CORE/src/Http/ActionDispatcher.php, line 117
Cake\Http\ActionDispatcher::dispatch() - CORE/src/Http/ActionDispatcher.php, line 96
Cake\Routing\Dispatcher::dispatch() - CORE/src/Routing/Dispatcher.php, line 60
[main] - ROOT/webroot/index.php, line 37
```